src/models.jl

```diff
@@ -1,6 +1,74 @@
 _maybe_fill(x, n) = x
 _maybe_fill(x::Number, n) = fill(x, n)
 
+function tight_binding(g::AbstractGraph; t=1.0, tp=0.0, h::Union{<:Real,Vector{<:Real}}=0)
+  h = _maybe_fill(h, nv(g))
+  ℋ = OpSum()
+  if !iszero(t)
```

Comment:
It is not necessary, and we should actually test that having terms with zero coefficients doesn't change the bond dimension. However, as a general rule (this is in `src/models.jl`), I am in favor of having this check because it lessens the computational expense of constructing the TTN (in the current implementation).

Reply:
Sounds good, we can reassess later.